diagnosisApplication/src/main/interface_adapter/diagnosis/DiagnosisState.java

```diff
@@ -8,134 +8,41 @@ public class DiagnosisState {
 
-    private  Integer diagnosis1Ranking = null;
-    private  String diagnosis1Name = null;
-    private  Integer diagnosis1Accuracy = null;
-    private  String diagnosis1Icd = null;
-    private  String diagnosis1ProfName = null;
-    private List<String> diagnosis1Specializations = null;
+    private HashMap<String, Object> diagnosis1;
+    private HashMap<String, Object> diagnosis2;
+    private HashMap<String, Object> diagnosis3;
 
-    private  Integer diagnosis2Ranking = null;
-    private  String diagnosis2Name = null;
-    private  Integer diagnosis2Accuracy = null;
-    private  String diagnosis2Icd = null;
-    private  String diagnosis2ProfName = null;
-    private List<String> diagnosis2Specializations = null;
-
-    private  Integer diagnosis3Ranking = null;
-    private  String diagnosis3Name = null;
-    private  Integer diagnosis3Accuracy = null;
-    private  String diagnosis3Icd = null;
-    private  String diagnosis3ProfName = null;
-    private List<String> diagnosis3Specializations = null;
-
-    public DiagnosisState() {
-
-    }
+    public DiagnosisState() {}
 
     public void setDiagnosis1(HashMap<String, Object> diagnosis1) {
-        this.diagnosis1Ranking = 1;
-        this.diagnosis1Name = (String) diagnosis1.get("name");
-        this.diagnosis1Accuracy = (Integer) diagnosis1.get("accuracy");
-        this.diagnosis1Icd = (String) diagnosis1.get("icd");
-        this.diagnosis1ProfName = (String) diagnosis1.get("profName");
-        this.diagnosis1Specializations = (List<String>) diagnosis1.get("specializations");
+        this.diagnosis1 = diagnosis1;
         numDiagnoses += 1;
     }
 
     public void setDiagnosis2(HashMap<String, Object> diagnosis2) {
-        this.diagnosis2Ranking = 2;
-        this.diagnosis2Name = (String) diagnosis2.get("name");
-        this.diagnosis2Accuracy = (Integer) diagnosis2.get("accuracy");
-        this.diagnosis2Icd = (String) diagnosis2.get("icd");
-        this.diagnosis2ProfName = (String) diagnosis2.get("profName");
-        this.diagnosis2Specializations = (List<String>) diagnosis2.get("specializations");
+        this.diagnosis2 = diagnosis2;
         numDiagnoses += 1;
     }
 
     public void setDiagnosis3(HashMap<String, Object> diagnosis3) {
-        this.diagnosis3Ranking = 3;
-        this.diagnosis3Name = (String) diagnosis3.get("name");
-        this.diagnosis3Accuracy = (Integer) diagnosis3.get("accuracy");
-        this.diagnosis3Icd = (String) diagnosis3.get("icd");
-        this.diagnosis3ProfName = (String) diagnosis3.get("profName");
-        this.diagnosis3Specializations = (List<String>) diagnosis3.get("specializations");
+        this.diagnosis3 = diagnosis3;
         numDiagnoses += 1;
     }
 
     public int getNumDiagnoses() {
         return numDiagnoses;
     }
 
-    public Integer getDiagnosis1Ranking() {
-        return diagnosis1Ranking;
+    public Object getDiagnosis1(String key) {
+        return diagnosis1.get(key);
```

Comment:
Setting up getters to access each diagnosis hash map so that front end does not directly interact with back end, and clean architecture is followed.